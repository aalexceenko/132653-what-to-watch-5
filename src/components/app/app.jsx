import React from "react";
import GeneralPage from "../general-page/general-page";
import PropTypes from "prop-types";
import {Switch, Route, BrowserRouter} from "react-router-dom";
import AddReviewScreen from "../add-review-screen/add-review-screen";
import FilmScreen from "../film-screen/film-screen";
import LoginScreen from "../login-screen/login-screen";
import MyListScreen from "../my-list-screen/my-list-screen";
import Player from "../player/player";
import {filmType} from '../../types/film';
import {reviewType} from '../../types/review';
import withPlayer from "../../hocs/with-player/with-player";

const PlayerWrapped = withPlayer(Player);

const AppRoute = {
  DEFAULT: `/`,
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
};

const App = ({films, reviews}) => {

  return (

    <BrowserRouter>
      <Switch>
        <Route exact path={AppRoute.DEFAULT}
          render={({history}) => (
            <GeneralPage
              films={films}
              reviews={reviews}
              onFilmCardClick={(id) => history.push(`/films/${id}`)}
              handleButtonPlayVideo={(id) => history.push(`/player/${id}`)}
            />
          )}
        />
        <Route exact path={AppRoute.LOGIN}>
          <LoginScreen />
        </Route>
        <Route exact path={AppRoute.MY_LIST}>
          <MyListScreen />
        </Route>
        <Route exact path="/films/:id"
          render={({history, match}) => (
            <FilmScreen
              films={films}
              reviews={reviews}
              match={match}
              onFilmCardClick={(id) => history.push(`/films/${id}`)}
              handleButtonPlayVideo={(id) => history.push(`/player/${id}`)}
            />
          )}
        />
        <Route exact path="/films/:id/review"
          render={({match}) => (
            <AddReviewScreen films={films} reviews={reviews} match={match}/>
          )}
        />
        <Route exact path="/player/:id"
          render={({match, history}) => (
            <PlayerWrapped films={films} match={match} handlePlayerExitClick={() => history.goBack()} />
          )}
        />
      </Switch>
    </BrowserRouter>
  );
};

App.propTypes = {
  films: PropTypes.arrayOf(filmType).isRequired,
  reviews: PropTypes.arrayOf(reviewType).isRequired,
};

export default App;
