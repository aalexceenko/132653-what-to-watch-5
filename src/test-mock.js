import {getFilmRank} from "./utils";

export const VISIBLE_FILMS_COUNT = 8;
export const ID = 1;
export const VIDEO_CURRENT_TIME = 0;
export const PROGRESS_POSITION = 0;
export const AUTHORIZATION_STATUS = `AUTH`;

export const IS_VIDEO = true;
export const IS_PLAYING = true;
export const IS_IN_MY_LIST = false;
export const TAB = `details`;
export const PREVIEW = `img/aviator.jpg`;
export const PREVIEW_VIDEO = `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`;
export const TITLE = `Bohemian Rhapsody`;
export const ACTIVE_GENRE = `Comedy`;
export const GENRES = [`Comedy`, `Crime`, `Documentary`, `Drama`, `Horror`, `Kids & Family`];
export const MATCH = {
  params: {
    id: 1
  }
};

export const FILMS = [
  {
    description: `A young man who was sentenced to seven years`,
    rating: 3.6,
    director: `Nicolas Winding Refn`,
    genre: `Action`,
    actors: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    id: 1,
    title: `Bronson`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    previewVideo: `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    trailer: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    votes: 109661,
    runtime: 92,
    year: 2008,
    ratingText: getFilmRank(3.6),
    myList: false,
    comments: false,
  },
  {
    description: `A young man who was`,
    rating: 3.6,
    director: `Nicolas Winding Refn`,
    genre: `Action`,
    actors: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    id: 2,
    title: `Bronson`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    previewVideo: `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    trailer: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    votes: 109661,
    runtime: 92,
    year: 2008,
    ratingText: getFilmRank(3.6),
    myList: false,
    comments: false,
  },
];

export const FILM = {
  description: `A young man who was sentenced to seven years`,
  rating: 3.6,
  director: `Nicolas Winding Refn`,
  genre: `Action`,
  id: 3,
  title: `Bronson`,
  actors: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
  poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
  backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
  previewVideo: `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  trailer: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  votes: 109661,
  runtime: 92,
  year: 2008,
  ratingText: getFilmRank(3.6),
  myList: false,
  comments: false,
};

export const REVIEWS = [
  {
    rating: 3.3,
    date: `2020-10-09T13:38:44.769Z`,
    user: {
      name: `Christina`,
      id: 1,
    },
    comment: `This film is an experience and i has already seen it 4 times`,
  },
  {
    rating: 3.3,
    date: `2020-10-09T13:38:44.769Z`,
    user: {
      name: `Christina`,
      id: 2,
    },
    comment: `This film is an experience and i has already seen it 4 times`,
  }
];
