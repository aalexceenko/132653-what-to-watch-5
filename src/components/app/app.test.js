import React from "react";
import renderer from "react-test-renderer";
import {App} from "./app";
import {FILMS, FILM} from "../../test-mock";

jest.mock(`../general-page/general-page`, () => `GeneralPage`);
it(`Should App render correctly`, () => {
  const tree = renderer
    .create(
        <App
          films={FILMS}
          filmPromo={FILM}
        />

    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
