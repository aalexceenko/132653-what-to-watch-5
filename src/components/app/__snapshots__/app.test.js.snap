// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should App render correctly 1`] = `
<GeneralPage
  filmPromo={
    Object {
      "actors": Array [
        "Tom Hardy",
        "Kelly Adams",
        "Luing Andrews",
      ],
      "backgroundImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg",
      "comments": false,
      "description": "A young man who was sentenced to seven years",
      "director": "Nicolas Winding Refn",
      "genre": "Action",
      "id": 3,
      "myList": false,
      "poster": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg",
      "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg",
      "previewVideo": "https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
      "rating": 3.6,
      "ratingText": "Normal",
      "runtime": 92,
      "title": "Bronson",
      "trailer": "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
      "votes": 109661,
      "year": 2008,
    }
  }
  films={
    Array [
      Object {
        "actors": Array [
          "Tom Hardy",
          "Kelly Adams",
          "Luing Andrews",
        ],
        "backgroundImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg",
        "comments": false,
        "description": "A young man who was sentenced to seven years",
        "director": "Nicolas Winding Refn",
        "genre": "Action",
        "id": 1,
        "myList": false,
        "poster": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg",
        "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg",
        "previewVideo": "https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
        "rating": 3.6,
        "ratingText": "Normal",
        "runtime": 92,
        "title": "Bronson",
        "trailer": "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
        "votes": 109661,
        "year": 2008,
      },
      Object {
        "actors": Array [
          "Tom Hardy",
          "Kelly Adams",
          "Luing Andrews",
        ],
        "backgroundImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg",
        "comments": false,
        "description": "A young man who was",
        "director": "Nicolas Winding Refn",
        "genre": "Action",
        "id": 2,
        "myList": false,
        "poster": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg",
        "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg",
        "previewVideo": "https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
        "rating": 3.6,
        "ratingText": "Normal",
        "runtime": 92,
        "title": "Bronson",
        "trailer": "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
        "votes": 109661,
        "year": 2008,
      },
    ]
  }
  handleButtonPlayVideo={[Function]}
  onFilmCardClick={[Function]}
/>
`;
